src/thunderbots/software/util/parameter/parameter.h

```diff
@@ -76,6 +77,36 @@ class Parameter
         // if the parameter hasn't been registered yet, return default value
         return this->value_;
     }
+
+    /**
+     * Returns true if the value of this parameter has been updated, either manually or
+     * from an automatic dynamic reconfigure message. Once this value is read, the return
+     * value is reset to false until the parameter is updated again. This makes the
+     * assumption that this function is only being called once / being called in one place
+     * per parameter, otherwise subsequent calls would not detect the update.
+     *
+     * @return true if the value of this parameter has been updated since this function
+     * was last called, and false otherwise
+     */
+    const bool valueUpdated()
+    {
+        bool ret = this->value_changed;
+
+        // get the value from the parameter in the registry
+        if (Parameter<T>::getMutableRegistry().count(this->name_))
+        {
+            auto& param_in_registry = Parameter<T>::getMutableRegistry().at(this->name_);
+            ret                     = param_in_registry->value_changed;
+            param_in_registry->value_changed = false;
+        }
+        else
+        {
+            value_changed = false;
```

Comment:
Only the parameters in the registry are updated, so this `else` block might be unnecessary as parameters are added to the registry when they are constructed.

Reply:
No longer relevant since I reverted the parameter changes.